virttest/storage.py

```diff
@@ -336,51 +352,91 @@ def save_to_file(self):
             fd.write(self.data)
 
-
-class ImageAccessInfo(object):
-    """
-    iscsi image access: initiator + StorageAuth(u/p)
-    ceph image access: StorageAuth(u/p)
-    """
-    def __init__(self, image, data, data_format, storage_type, initiator=None):
-        self.image = image
-        self.storage_type = storage_type
-        auth_type = None
-        if self.storage_type == 'iscsi-direct':
-            auth_type = 'chap'
-            self.iscsi_initiator = initiator
-        elif self.storage_type == 'ceph':
-            auth_type = 'cephx'
-        self.auth = StorageAuth(self.image, data, data_format,
-                                auth_type) if data is not None else None
-
     @classmethod
-    def access_info_define_by_params(cls, image, params):
+    def auth_info_define_by_params(cls, image, params):
+        """
+        :param image: image tag name
+        :param params: image specified parmas, i.e. params.object_params(image)
+        """
+        auth = None
+        storage_type = params.get("storage_type")
         enable_ceph = params.get("enable_ceph", "no") == "yes"
         enable_iscsi = params.get("enable_iscsi", "no") == "yes"
-        storage_type = params.get("storage_type")
-        access_info = None
 
         if enable_iscsi:
             if storage_type == 'iscsi-direct':
                 initiator = params.get('initiator')
                 data = params.get('chap_passwd')
-                data_format = params.get("data_format", "raw")
-                access_info = cls(image, data, data_format, storage_type,
-                                  initiator) if data or initiator else None
+                data_format = params.get('data_format', 'raw')
+                auth = cls(image, data, data_format, storage_type,
+                           initiator=initiator) if data or initiator else None
         elif enable_ceph:
             data = params.get('ceph_key')
-            data_format = params.get("data_format", "base64")
-            access_info = cls(image, data, data_format,
-                              storage_type) if data else None
+            data_format = params.get('data_format', 'base64')
+            auth = cls(image, data, data_format,
+                       storage_type) if data else None
+        return auth
+
+
+class ImageAccessInfo(object):
+    """
+    Access info to the logical image, which can include the network
+    storage image only, or the image and its backing images.
+    """
+    def __init__(self, image, image_auth, image_backing_auth):
+        """
+        :param image: image tag name
+        :param image_auth: StorageAuth object to access image itself
+        :param image_backing_auth: a dict({image: StorageAuth object}),
+                                   used for accessing the backing images
+        """
+        self.image = image
+        self.image_auth = image_auth
+        self.image_backing_auth = image_backing_auth
```

Comment:
Do we really need this?
It looks structure should be looks like:
````
image =  QemuStorage.QemuImg(...)
backing_image = QemuStorage.QemuImg(...)
backing_image.access_info = ImageAccessInfo.access_info_define_by_params(image, params)
image.backing = backing_image
```

Reply:
Backing files should not be distinguished, it is also a type of image file.

Reply:
Yes, you are right. But QemuImg now is more like a logical image, which includes the image itself as well as its backing image, and the backing image is simply described by 'some information'(like tag, filename...,) instead of an QemuImg object. 
So, according to the current design, ImageAccessInfo was also designed to a class which describes the logical image's access, i.e. the access info for the image itself as well as its backing images. ImageAccessInfo is used by qemu-img and qemu-kvm, to form the right command line, for qemu-img command, it's not always necessary to know all images(in image chain) access information, e.g. 'qemu-img info' without '--backing-chain' option only needs to know the access info for the image itself, while 'qemu-img info --backing-chain' needs not only the access info for the image itself but also its backing images' access info, so in ImageAccessInfo, we distinguished the access info for the image and its backing.